src/ORM/Table.php

```diff
@@ -843,9 +843,11 @@ public function behaviors(): BehaviorRegistry
     /**
      * Get a behavior from the registry.
      *
-     * @template TName of key-of<TBehaviors>
-     * @param TName $name The behavior alias to get from the registry.
-     * @return TBehaviors[TName]
+     * @param string $name The behavior alias to get from the registry.
+     * @return \Cake\ORM\Behavior
+     * @phpstan-template TName of key-of<TBehaviors>
```

Comment:
I was only talking about param/return, as those are needed for most IDEs to understand the basic input/output.
`@template` can stay IMO.

But as for it working:
Even with your first commit I still always get
```
PHPDoc tag @extends contains generic type 
Cake\ORM\Table<array<string, Cake\ORM\Behavior\TimestampBehavior>> 
but class Cake\ORM\Table is not generic.  
```

Reply:
were you using the correct branch?